src/Composer/EventDispatcher/EventDispatcher.php

```diff
@@ -350,7 +350,7 @@ protected function doDispatch(Event $event)
                         }
                         // match somename (not in quote, and not a qualified path) and if it is not a valid path from CWD then try to find it
                         // in $PATH. This allows support for `@php foo` where foo is a binary name found in PATH but not an actual relative path
-                        $matched = Preg::isMatchStrictGroups('{^[^\'"\s/\\\\]+}', $pathAndArgs, $match);
+                        $matched = Preg::isMatchStrictGroups('{^[^-\'"\s/\\\\]+}', $pathAndArgs, $match);
```

Comment:
Let's rather just prevent - from being present as first char, because after that something like `foo-bar` should be fine.

```suggestion
                        $matched = Preg::isMatchStrictGroups('{^(?!-)[^\'"\s/\\\\]+}', $pathAndArgs, $match);
```